### PR TITLE
Ts tree cursor copy

### PR DESCRIPTION
**Pull request by zajac**
It is impossible to implement this function in user-space using existing functions without breaking encapsulation of TSTreeCursor.
Sometimes it is useful to save current position in tree and ts_tree_cursor_new will not do the trick because cursor created for some node will not be able to ascend to it's parent with ts_tree_cursor_goto_parent

### CODE REVIEW

**Comment by zajac** [comment]
@maxbrunsfeld  Here it is

File: lib/src/tree_cursor.c [hunk 1]
```diff
@@ -291,3 +291,12 @@ const char *ts_tree_cursor_current_field_name(const TSTreeCursor *_self) {
...
   }
 }
+
+TSTreeCursor ts_tree_cursor_copy(const TSTreeCursor *_cursor) {
+    const TreeCursor *cursor = (const TreeCursor *)_cursor;
+    TSTreeCursor res = {NULL, NULL, {0, 0}};
+    TreeCursor *copy = (TreeCursor *)&res;
+    copy->tree = cursor->tree;
+    array_push_all(&copy->stack, &cursor->stack);
+    return res;
```
Comment (maxbrunsfeld) [inline lib/src/tree_cursor.c]: It looks like this has a different indentation than the other functions. Could you adjust it to match?

Comment (zajac) [reply lib/src/tree_cursor.c]: BTW shouldn't we increment ref-count of the tree here?

Comment (zajac) [reply lib/src/tree_cursor.c]: The cursor doesn't seem to retain it at all. Probably that's fine

**Pull request merged**